stout/flags.h

```diff
@@ -107,16 +131,45 @@ class Parser {
   // Optional for including all environment variables for parsing
   // with specific prefix.
   std::optional<std::string> environment_variable_prefix_;
+
+  // Map from nested parser to it's all possible names. Depending on the
+  // subcommand defined by the users we can grab specific parser and do
+  // parsing for the next arguments in the command line.
+  std::map<std::unique_ptr<Parser>, std::set<std::string>> nested_parsers_;
+
+  // Helper for saving the message for which flags or pos args are going
+  // to be parsed.
+  google::protobuf::Message* message_for_parsing_ = nullptr;
+
+  // Flag will be set to 'true' if the nested parser was created.
+  bool is_nested_ = false;
 };
 
 ////////////////////////////////////////////////////////////////////////
 
+
 template <typename Flags>
 class ParserBuilder {
  public:
   ParserBuilder(Flags* flags)
     : flags_(flags) {
-    parser_.AddAllOrExit(flags_);
+    // When we construct top-level parser, we need to specify
+    // 'Parser::message_for_parsing_' for parsing top-level flags
+    // and in order to easily grab 'google::protobuf::Message*'
+    // pointer for nested parsers too.
+    // NOTE: 'Parser::message_for_parsing_' will be specified for
+    // nested parsers only in runtime, when we actually do the par-
+    // sing because of 'oneof' implementation (we can not store at
+    // a time 'google::protobuf::Message*' pointers for all 'oneof'
+    // fields).
+    parser_.message_for_parsing_ = flags_;
```

Comment:
I don't understand this ... why do you save the message here as well as reset it at runtime?